cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -432,30 +477,31 @@ Taking our first example and modifying it, you could use code such as::
 where ``MyPredicate`` is a public class which implements the ``Predicate`` interface as::
 
   public static class MyPredicate implements Predicate<WorkflowContext> {
-
     @Override
     public boolean apply(@Nullable WorkflowContext input) {
-       if (input == null) {
-          return false;
-       }
-       String flattenCounterName = "org.apache.hadoop.mapreduce.TaskCounter.MyCustomCounters";
-       Value tokenValue = input.getToken().get(flattenCounterName, "BuildProductProfile", WorkflowToken.Scope.SYSTEM);
-       if (tokenValue == null) {
-          return false;
-       }
-       if (tokenValue.getAsInt() > 0) {
-         return true;
-       }
-       return false;
+      if (input == null) {
```

Comment:
When the MapReduce program is called directly, without the Workflow, then the WorkflowContext would be null. This is documented here https://github.com/caskdata/cdap/blob/release/3.3/cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java#L200. However for custom actions it will not be null. 

Added documentation for the same in rst file as well.